torch/_torch_docs.py

```diff
@@ -3152,7 +3152,24 @@ def merge_dicts(*dicts):
            r"""
 flatten(input, start_dim=0, end_dim=-1) -> Tensor
 
-Flattens a contiguous range of dims in a tensor.
+By default, flattens :attr:`input` into a one-dimensional tensor with the same number of elements.
```

Comment:
I had the opportunity to think about this more, and we could also use language like:

"By default, rearranges the elements of :attr:\`input` into a one-dimensional tensor while preserving their order"

"Rearrange" seems helpful to me because it implies the same data is used. "Preserving their order" is vague but maybe the best way to articulate the relationship between elements? The number of elements is OK but doesn't speak to the arrangement of the elements.

Reply:
Its interesting to think in this direction. Could even just say 'reshape'?

Its good to add "preserve their order". Maybe even further specify it as "row-major order"?

Reply:
Thoughts on removing "By default"? Prefixing with this condition seems to imply that order may not preserved otherwise.

Reply:
Reshape sounds good. I don't know if the ordering is always "row-major" if the tensor is not contiguous, though.

A way to workaround "by default" might be to use a construction like this:

"Flattening a tensor :attr:\`input\` ..."

And then later:

"If start_dim or end_dim are specified then only those dimensions and the dimensions between them are flattened."

?

Reply:
To make the first sentence explain what the function does this could also begin with:

"Flattens :attr:\`input\`  or, if start_dim or end_dim are specified, a range of :attr:\`inputs\` dimensions."

Then next paragraph defines "flattening" like the above.